### PR TITLE
Feat/check path conflict interval

### PR DESCRIPTION
**Pull request by ArthurKChan**
- Implements a interval every 12 seconds to look for path conflicts and reroute if there is a conflict
- Implements communications if path conflict comes up
- Adds utils
  - getFlightData
  - removeFlightPath
  - updateFlightPath
  - checkForPathConflicts

cloese #27

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -17,7 +17,30 @@ var tSay = function(msg) {
...
+                <speed>:,
+                <prevPathPtInd>:,
+                <dist>:,
+                <statusCode>:,
+                <transcript>:
+              },
+              ...
+  */
+};
+var pendingPathUpdates = {
```
Comment (ArthurKChan) [inline server/server.js:39]: local storage for pending flight path updates

File: server/server.js [hunk 2]
```diff
@@ -6,6 +6,11 @@ proj4.defs("ESRI:102243","+proj=lcc +lat_1=37.06666666666667 +lat_2=38.433333333
 var port = process.env.PORT || 8080;
 var io = require('socket.io')(port);
```
Comment (ArthurKChan) [inline server/server.js:8]: Global Variables for interval timers

File: server/server.js [hunk 3]
```diff
@@ -120,21 +144,76 @@ io.on('connection', function(socket){
...
+    .then(function(pathData){
+      var originalPath = JSON.parse(pathData[0].path_geom);
+      var newPath = originalPath.coordinates.slice(0,pathUpdate.timeBufPrevPtInd).concat(pathUpdate.path);
+      var newPathGeoJSON = JSON.stringify({"type":"LineString","coordinates":newPath});
+      return utils.updateFlightPath(msg.callSign, newPathGeoJSON)
+    })
+    .then(function(result){
+      console.log('updated database')
+      console.log(result)
+      delete pendingPathUpdates[msg.callSign];
```
Comment (ArthurKChan) [inline server/server.js:164]: deletes the path update from pendingPathUpdates after the drone acknowledges it updated its path

**Comment by dennisgl** [comment]
:+1:

**Pull request merged**